### PR TITLE
Add Basic Benchmark Code And Fix Performance Bug

### PR DESCRIPTION
**Pull request by isomarcte**
* Performance bug 
    * The generated code was creating a `def` for `mapped`, when it should have been a `lazy val`. This caused us to pay the initialization cost on each invocation.

### CODE REVIEW

**Review by armanbilge — APPROVED** [review]
👍 so how do we compare to icu4j at the moment?

**Comment by isomarcte** [comment]
I'm running a fresh benchmark now. jmh says it will be done in about 30 minutes. I'll post when it is done. Smaller benchmarks I ran earlier, after I fixed this bug, were very promising.

File: project/UTS46CodeGen.scala [hunk 1]
```diff
@@ -797,7 +796,7 @@ object UTS46CodeGen {
           q"private final def mapped1 = ${singleMappingToTrees(mapped1)}",
           q"private final def mapped2 = ${singleMappingToTrees(mapped2)}",
           q"private final def mapped3 = ${singleMappingToTrees(mapped3)}",
-          q"protected override final def mapped = mapped0 ++ mapped1 ++ mapped2 ++ mapped3"
+          q"protected override final lazy val mapped = mapped0 ++ mapped1 ++ mapped2 ++ mapped3"
```
[the remark this answers is not in the document]
Comment (armanbilge) [reply project/UTS46CodeGen.scala:799]: Btw, `lazy val`s add a synchronization thing. What we might want here is an "unsafe" lazy that may initialize multiple times if hit concurrently, but less cost for subsequent access. Scala 3 offers this out-of-the-box, but on Scala 2 we have to implement this ourselves.

**Comment by isomarcte** [comment]
@armanbilge 

```
[info] Benchmark                Mode  Cnt       Score      Error  Units
[info] ICU4J.icu4jUTS46        thrpt   25  148022.999 ±  728.902  ops/s
[info] ICU4J.icu4jUTS46ASCII   thrpt   25  199360.511 ± 1889.718  ops/s
[info] ICU4J.idna4sUTS46       thrpt   25  196386.913 ± 1652.306  ops/s
[info] ICU4J.idna4sUTS46ASCII  thrpt   25  525445.483 ± 7324.678  ops/s
```

For arbitrary `String` values over the whole Unicode code point domain, idna4s runs ~132% faster than icu4j. 

For arbitrary _ASCII only_ `String` values idna4s runs ~263% faster than icu4j.

**Comment by armanbilge** [comment]
Holy crap 😮

File: project/UTS46CodeGen.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (isomarcte) [reply project/UTS46CodeGen.scala:799]: Sorry, I missed this comment. That's a good point, I'll loop back to this.

**Pull request merged**